waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -964,6 +973,25 @@
       has-reverse-proxy?
       (attach-envoy-sidecar reverse-proxy service-description base-env service-port port0))))
 
+(defn default-pdb-spec-builder
+  "Factory function which creates a Kubernetes PodDisruptionBudget spec for the given ReplicaSet."
+  [pdb-api-version rs-spec replicaset-uid]
+  (let [rs-api-version (get rs-spec :apiVersion)
+        rs-kind (get rs-spec :kind)
+        rs-name (get-in rs-spec [:metadata :name])
+        rs-selector (get-in rs-spec [:spec :selector])]
+    {:apiVersion pdb-api-version
+     :kind "PodDisruptionBudget"
+     :metadata {:name (str rs-name "-pdb")
```

Comment:
Should we add something random to the name?

If an RS gets deleted and then a replacement is created before the k8s gc runs, then the old RS's PDB might still be around, and cause a conflict error when we try to post a new PDB with the same name.

Reply:
My understanding of `blockOwnerDeletion true` means that the PDB needs to be deleted before the ReplicaSet deletion is complete and any recreation triggered.

> Once the garbage collector has deleted all "blocking" dependents (objects with ownerReference.blockOwnerDeletion=true), it deletes the owner object.

https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/

Reply:
We use Background deletion, not Foreground:

https://github.com/twosigma/waiter/blob/eeb0e6fb98a4e13c0e6ef2104f5390c0cf1cb57d/waiter/src/waiter/scheduler/kubernetes.clj#L553-L554

Your quote is from the Foreground section. The rules for Background are different:

>In background cascading deletion, Kubernetes deletes the owner object immediately and the garbage collector then deletes the dependents in the background.

https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#background-cascading-deletion